From: Chad Monroe <chad@monroe.io>
Date: Wed, 25 Feb 2026 19:07:10 -0800
Subject: [PATCH] wifi-scripts: disable background radar in AP+STA mode

In AP+STA mode the AP must follow the STA's channel. Background radar
CAC can trigger a channel switch to a DFS-cleared channel that the STA
can't follow, causing tear down of colocated VAPs.

Detect a STA interface in hostapd.setup() and skip enable_background_radar
in the device config generator (usr/share/ucode/wifi/hostapd.uc).

Signed-off-by: Chad Monroe <chad@monroe.io>
---
--- a/src/usr/share/ucode/wifi/hostapd.uc
+++ b/src/usr/share/ucode/wifi/hostapd.uc
@@ -520,7 +520,7 @@ function generate(config) {
 	append_vars(config, [ 'rssi_reject_assoc_rssi', 'rssi_reject_assoc_timeout', 'rssi_ignore_probe_request', 'iface_max_num_sta' ]);
 
 	/* ACS / Radar*/
-	if (!phy_features.radar_background || config.band != '5g')
+	if (!phy_features.radar_background || config.band != '5g' || config.has_sta)
 		delete config.enable_background_radar;
 	else
 		set_default(config, 'enable_background_radar', false);
@@ -578,6 +578,10 @@ export function setup(data) {
 
 	data.config.phy = data.phy;
 
+	for (let k, interface in data.interfaces)
+		if (interface.config.mode == 'sta')
+			data.config.has_sta = true;
+
 	generate(data.config);
 
 	if (data.config.num_global_macaddr)
